public_html/admin/controller/pages/sale/customer.php

```diff
@@ -1084,9 +1084,10 @@ public function actonbehalf()
             // and then to storefront because cross-domain restriction for session cookie
             /** @var ModelSettingStore $mdl */
             $mdl = $this->loadModel('setting/store');
-            $store_settings = $mdl->getStore($this->session->data['current_store_id']);
+            $store_id = $this->getStoreId();
```

Comment:
what for this code? 
We have no assignment customer-to-store yet.

Reply:
that is for obtaining the correct store id. please test actonbehalf in a clean v1.4.2 to see the issue.

 i am redirected to the multistore admin url, but when i login, i get invalid token. tried several times but still the same